### PR TITLE
[Merged by Bors] - Unban peers at the swarm level when purged

### LINKED ISSUE #2840: Removed banned peers from the swarm
## Description

If we have excess banned peers we remove old ones from being tracked but potentially do not remove them from being banned in the swarm. The banned peers removed from the peer-db should be unbanned by the swarm

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed
#2840

### CODE REVIEW

File: beacon_node/lighthouse_network/src/peer_manager/peerdb.rs [hunk 1]
```diff
@@ -709,6 +709,7 @@ impl<TSpec: EthSpec> PeerDB<TSpec> {
                     }
                     PeerConnectionStatus::Banned { .. } => {
                         error!(self.log, "Accepted a connection from a banned peer"; "peer_id" => %peer_id);
+                        // TODO: check if this happens and report the unban back
```
Comment (AgeManning) [inline beacon_node/lighthouse_network/src/peer_manager/peerdb.rs:712]: This shouldn't happen. But we should report it back for completeness, I think

**Review by AgeManning — COMMENTED** [review]
LGTM, minor comment

File: beacon_node/lighthouse_network/src/peer_manager/peerdb.rs (hunk 1 above)
Comment (divagant-martian) [reply beacon_node/lighthouse_network/src/peer_manager/peerdb.rs:712]: I checked what needs to be changed to address this and changes are quite a few. There are some things we could fix first to make this easier to fix. Would you mind if we add this as is?

**Review by AgeManning — APPROVED** [review]

**Pull request closed without merging**